simulator/virtual_machine_test.go

```diff
@@ -1820,6 +1820,149 @@ func TestVmSnapshot(t *testing.T) {
 	}
 }
 
+func TestVmSnapshotEx(t *testing.T) {
+	ctx := context.Background()
+
+	m := ESX()
+	defer m.Remove()
+	err := m.Create()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	s := m.Service.NewServer()
+	defer s.Close()
+
+	c, err := govmomi.NewClient(ctx, s.URL, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
```

Comment:
Instead of the boilerplate above, you can use the new `Test` helper:
```go
Test(func(ctx context.Context, c *vim25.Client) {
    ...
}, ESX())
```

Example: https://github.com/vmware/govmomi/blob/main/simulator/folder_test.go#L633